source/h2_stream.c

```diff
@@ -344,6 +357,17 @@ static void s_stream_destroy(struct aws_http_stream *stream_base) {
     aws_mutex_clean_up(&stream->synced_data.lock);
     aws_http_message_release(stream->thread_data.outgoing_message);
 
+    /* chuck the outgoing write on the pending list for unified clean up */
+    if (stream->thread_data.outgoing_write) {
+        aws_linked_list_push_front(&stream->synced_data.pending_write_list, &stream->thread_data.outgoing_write->node);
+    }
+    /* clean up any pending writes */
+    while (!aws_linked_list_empty(&stream->synced_data.pending_write_list)) {
```

Comment:
we might also want to purge this list in on_stream_complete. because the stream/connection might have died, so why keep holding onto these resources. especially since someone's callback-userdata might have a reference to the stream itself

look at what we did purging http1 chunks